include/plugin.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef GRAPH_PROTOTYPE_PLUGIN_H
+#define GRAPH_PROTOTYPE_PLUGIN_H
+
+#include <span>
+#include <string>
+
+#include <dlfcn.h>
+
+#include <graph.hpp>
+#include <node_registry.hpp>
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+#define GP_PLUGIN_CURRENT_ABI_VERSION 1
+
+struct gp_plugin_metadata {
+    std::string_view plugin_name;
+    std::string_view plugin_author;
```

Comment:
I think it's useful to have some generic information reachable from code to e.g show in a UI, but I think having the this could all be handled via the annotation interface of the blocks.

Reply:
I'd add this as part of the source code...?!?

Reply:
As a user, I get the binary (provided by my linux distro for example), how do I know the author, license, version?